arangod/Replication2/ReplicatedLog/Components/CompactionManager.cpp

```diff
@@ -126,16 +126,19 @@ void CompactionManager::triggerAsyncCompaction(
         // The participant might be gone
         LOG_CTX("d7724", DEBUG, self->loggerContext)
             << "error during compaction: " << storeRes.result();
-        auto cresult = CompactResult{
-            .error = std::move(storeRes).result().error(),
-            .stopReason =
-                CompactionStopReason{
-                    CompactionStopReason::StorageManagerNotReady{}},
-            .compactedRange = {}};
-        guard->status.lastCompaction->error = cresult.error;
+
+        result::Error err{storeRes.result().errorNumber(),
+                          storeRes.result().errorMessage()};
+        auto& compaction = guard->status.inProgress.emplace();
+        guard->status.lastCompaction = guard->status.inProgress;
+        guard->status.inProgress.reset();
+        compaction.error = std::move(err);
```

Comment:
This looks a little convoluted, maybe a c&p relic? Why do you emplace `inProgress`, then copy it to `lastCompaction`, then reset it again, and _then_ write the `.error` field? Isn't this writing in the middle of a  freshly reset optional?

The way I see it: If `inProgress` is there, use that. If it's not, construct a new one, but don't forget to set the `.time` field to `CompactionStatus::clock::now()`. E.g.
```suggestion
        if (guard->status.inProgress.has_value()) {
          guard->status.lastCompaction = guard->status.inProgress;
          guard->status.inProgress.reset();
        } else {
          guard->status.lastCompaction = Compaction();
          guard->status.lastCompaction.time = CompactionStatus::clock::now();
        }
        guard->status.lastCompaction.error = std::move(err);
```
(which suggests that now `Compaction::range` should be an optional, but let's keep it at that ;).

Reply:
Thanks, I simplified it in https://github.com/arangodb/arangodb/pull/20628/commits/f480fa0c1aa8c6cb6615763e7aafc7bf2e60b58d